jupyterhub/spawner.py

```diff
@@ -350,10 +350,24 @@ def _deprecate_percent_u(self, proposal):
         """
     ).tag(config=True)
 
+    _instances = {}
+    
     def __init__(self, **kwargs):
         super(Spawner, self).__init__(**kwargs)
         if self.user.state:
             self.load_state(self.user.state)
+    
+    def save_spawner(self, server_name):
+        Spawner._instances[server_name] = self
+        self.server_name = server_name
+
+    @classmethod
+    def get_spawners(cls):
+        return cls._instances
+    
+    @classmethod
+    def get_spawner(cls, server_name):
+        return cls._instances[server_name]
```

Comment:
> so you would rather prefer to move this implementation inside the user.py?

Yes. A user has spawners, so `spawners` should be an attribute of the User object.

> I was thinking that from a perspective of a service that would like to control the spawner then it would be hard to access Spawner's instances.

There aren't services in memory that have access to the Spawner class, so they wouldn't be able to access the instances class attribute.